File: src/lib/scala/Slate.Common/src/main/scala/slate/common/query/Condition.scala
package slate.common.query


class Condition(val field:Any, val comparison:String, val fieldValue:Any) extends ICondition {

  /**
    * string represention of condition
    * @return
    */
   def toStringQuery:String =
  {
    toStringQueryWithOptions(false, "[", "]")
  }


  /**
   * Returns a String representation of this instance.
   * @param surround : True to surround alias with text.
   * @param left     : Left surrounding text
   * @param right    : Right surrounding text
   * @return
   */
  def toStringQueryWithOptions(surround:Boolean=false, left:String="", right:String="") : String =
  {
    val fieldName = QueryEncoder.ensureField(this.field.toString)
    val col = if (surround) left + fieldName + right else fieldName
    val comp = QueryEncoder.ensureCompare(comparison)
    val fieldVal = this.fieldValue
    val result = QueryEncoder.convertVal(fieldVal)

    col + " " + comp + " " + result
  }
}

File: src/lib/kotlin/slatekit-query/src/main/kotlin/slatekit/query/Condition.kt
/**
 * <slate_header>
 * url: www.slatekit.com
 * git: www.github.com/code-helix/slatekit
 * org: www.codehelix.co
 * author: Kishore Reddy
 * copyright: 2016 CodeHelix Solutions Inc.
 * license: refer to website and/or github
 * about: A tool-kit, utility library and server-backend
 * mantra: Simplicity above all else
 * </slate_header>
 */

package slatekit.query

class Condition(@JvmField val field: Any,
                @JvmField val comparison: String,
                @JvmField val fieldValue: Any) : ICondition {

    /**
     * string represention of condition
     * @return
     */
    override fun toStringQuery(): String = toStringQueryWithOptions(false, "[", "]")

    /**
     * Returns a String representation of this instance.
     * @param surround : True to surround alias with text.
     * @param left : Left surrounding text
     * @param right : Right surrounding text
     * @return
     */
    fun toStringQueryWithOptions(surround: Boolean = false, left: String = "", right: String = ""): String {
        val fieldName = QueryEncoder.ensureField(this.field.toString())
        val col = if (surround) left + fieldName + right else fieldName
        val comp = QueryEncoder.ensureCompare(comparison)
        val fieldVal = this.fieldValue
        val result = QueryEncoder.convertVal(fieldVal)

        return "$col $comp $result"
    }
}
